src/System.Net.WebSockets.WebSocketProtocol/src/System/Net/WebSockets/ManagedWebSocket.netstandard.cs

```diff
@@ -0,0 +1,26 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+// NOTE: This file is shared between CoreFX and ASP.NET.  Be very thoughtful when changing it.
+
+namespace System.Net.WebSockets
+{
+    internal sealed partial class ManagedWebSocket : WebSocket
+    {
+        private Task ValidateAndReceiveAsync(Task receiveTask, byte[] buffer, CancellationToken cancellationToken)
+        {
+            if (receiveTask == null ||
+                (receiveTask.Status == TaskStatus.RanToCompletion &&
+                !(receiveTask is Task<WebSocketReceiveResult> wsrr && wsrr.Result.MessageType == WebSocketMessageType.Close)))
+            {
+                receiveTask = ReceiveAsyncPrivate<WebSocketReceiveResultGetter, WebSocketReceiveResult>(new ArraySegment<byte>(buffer), cancellationToken).AsTask();
```

Comment:
Why did you remove the special-casing of ReceiveAsyncPrivate returning an already successfully completed task?

Reply:
Nevermind; I was looking at the wrong method.